### PR TITLE
Fix `mm-snap build` command when CLI is installed globally

### LINKED ISSUE #206: Error when init mm-snap
Error when running `mm-snap init`

```
MetaMask Snaps: Initialize

package name: (example) 
version: (1.0.0) 
description: 
entry point: (index.js) 
test command: 
git repository: 
keywords: 
author: 
license: (ISC) 
About to write to /home/vid/Documents/Blockchain_LabUM/MetaMask/example/package.json:

{
  "name": "example",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}


Is this OK? (yes) yes

Init: Building 'snap.manifest.json'...

Use default Snap manifest values? (yes) yes
Using default values...

Init: Created 'snap.manifest.json'.
Init: Created 'index.js'.
Init: Created 'index.html'.
Init: Wrote 'snap.config.json' config file
events.js:187
      throw er; // Unhandled 'error' event
      ^

Error: Cannot find module 'babelify' from '/home/vid/Documents/Blockchain_LabUM/MetaMask/example'
    at /home/vid/.nvm/versions/node/v12.11.0/lib/node_modules/@metamask/snaps-cli/node_modules/resolve/lib/async.js:146:35
    at processDirs (/home/vid/.nvm/versions/node/v12.11.0/lib/node_modules/@metamask/snaps-cli/node_modules/resolve/lib/async.js:299:39)
    at isdir (/home/vid/.nvm/versions/node/v12.11.0/lib/node_modules/@metamask/snaps-cli/node_modules/resolve/lib/async.js:306:32)
    at /home/vid/.nvm/versions/node/v12.11.0/lib/node_modules/@metamask/snaps-cli/node_modules/resolve/lib/async.js:34:69
    at FSReqCallback.oncomplete (fs.js:169:21)
Emitted 'error' event on Browserify instance at:
    at /home/vid/.nvm/versions/node/v12.11.0/lib/node_modules/@metamask/snaps-cli/node_modules/browserify/index.js:375:34
    at /home/vid/.nvm/versions/node/v12.11.0/lib/node_modules/@metamask/snaps-cli/node_modules/resolve/lib/async.js:148:17
    at processDirs (/home/vid/.nvm/versions/node/v12.11.0/lib/node_modules/@metamask/snaps-cli/node_modules/resolve/lib/async.js:299:39)
    [... lines matching original stack trace ...]
    at FSReqCallback.oncomplete (fs.js:169:21) {
  code: 'MODULE_NOT_FOUND'
}
```

Version: 0.8.0
Installation: `yarn global add @metamask/snaps-cli`

### PR DESCRIPTION
**Pull request by rekmarks**
As reported in #206, `mm-snap build` fails with an error reporting that `babelify` cannot be found if the CLI is installed globally. In local testing, I can reproduce this behavior with both `yarn` and `npm`. The cause of this error has something to do with the way Browserify and / or Babel resolve dynamically imports their various transforms, presets, and plugins. By importing all of these statically using `require` instead of passing their names as string literals, the "module not found" error disappears.

I considered bundling the CLI as an alternative, but that appears to be difficult precisely due to the dynamic imports that caused the problem in the first place.

To test this manually, build this branch, run `npm pack`, install the resulting tarball globally using `npm install -g ./path-to-tarball.tgz`, find your global npm binary path using `npm bin -g`, and then invoke the `mm-snap` file in there. Note that you must not have a different version of `@metamask/snaps-cli` installed globally at the same time for the same Node version.

Fixes #206

### CODE REVIEW

**Comment by rekmarks** [comment]
Pending #213

File: packages/snaps-cli/src/cmds/build/bundle.test.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { bundle } from './bundle';
 import * as bundleUtils from './bundleUtils';
 
 jest.mock('browserify');
+jest.mock('babelify', () => 'mockBabelify');
```
Comment (rekmarks) [inline packages/snaps-cli/src/cmds/build/bundle.test.ts:7]: The default export of `babelify` is actually a function, but we just need to know that whatever is exported under that name is passed as the first argument to the transform in the test cases in this file.

File: packages/snaps-cli/src/cmds/build/bundle.ts [hunk 2]
```diff
@@ -3,6 +3,10 @@ import { YargsArgs } from '../../types/yargs';
 import { TranspilationModes } from '../../builders';
 import { createBundleStream, closeBundleStream } from './bundleUtils';
 
+// We need to statically import all Browserify transforms and all Babel presets
```
Comment (hmalik88) [inline packages/snaps-cli/src/cmds/build/bundle.ts:6]: typo here

File: packages/snaps-cli/src/cmds/build/bundle.ts [hunk 3]
```diff
@@ -3,6 +3,10 @@ import { YargsArgs } from '../../types/yargs';
...
+// presets and plugins, and calling `require` is the sanest way to do that.
```
Comment (rekmarks) [inline packages/snaps-cli/src/cmds/build/bundle.ts:7]:
```suggestion
// and plugins, and calling `require` is the sanest way to do that.
```

File: packages/snaps-cli/src/cmds/build/bundle.ts (hunk 2 above)
Comment (rekmarks) [reply packages/snaps-cli/src/cmds/build/bundle.ts:6]: Fixed!

**Review by shanejonas — APPROVED** [review]

**Review by hmalik88 — APPROVED** [review]

**Pull request merged**